mitodl/mit-open
# Type-specific api endpoints for videos and video playlists

### What are the relevant tickets?
Closes #592

### Description (What does it do?)
Adds the following url endpoints:

/api/v1/videos/
/api/v1/video_playlists/
/api/video_playlists/<resource_id>/items/


### How can this be tested?
- If you don't already have videos and playlists in your local instance:
  - Set `YOUTUBE_CONFIG_URL=https://raw.githubusercontent.com/mitodl/open-video-data/mb/one_file/youtube/channels.yaml`
  - Set `YOUTUBE_DEVELOPER_KEY=<RC value>`
  - Run `./manage.py backpopulate_youtube_data fetch`
- Try the following urls, they should return appropriate data:
  - http://localhost:8063/api/v1/videos/  - list of videos
  - http://localhost:8063/api/v1/videos/video_id/ - details for a specific video
  - http://localhost:8063/api/v1/video_playlists/ - list of video playlists
  - http://localhost:8063/api/v1/video_playlists/playlist_id/ - details for a specific playlist
  - http://localhost:8063/api/v1/video_playlists/playlist_id/items/ - videos in the playlist
